### ISSUE TITLE
Devices not discovered

### ISSUE THREAD

**Issue by mnielsm** [ruebox/openhab2-addons#43]
Hi,
Good work so far with the openhab - free@home connection.
Unfortunately in my installation only one of all my devices is discovered.
Discovered:
- Thermostat (deviceID 1004)
Not discovered: 
- 2/2 sensor/rocker (deviceID 1010)
- Motion sensor (deviceID 100A)
- 1/1 sensor/dimmer (deviceID 1017)
- 1/1 sensor/rocker (deviceID 100C)
- Wireless 1/1 sensor/dimmer (deviceID 2039)
- 1 sensor (deviceID 1000)

I am trying to figure out if I am able to add these to the system myself but have to look into the code to be able to do that.
If anybody is able to help please do.

Attached is my (stripped down, user, backups and floorplan removed) getall.xml file

[getAll_backup_users removed.zip](https://github.com/ruebox/openhab2-addons/files/2343396/getAll_backup_users.removed.zip)

**Comment by hendrik-d** [comment]
Hello! Did you find any solution regarding the undiscovered devices? In my installation only rollershutters, scenes and switches are discovered but no lights/dimmers…

**Comment by ruebox** [comment]
You are right, the listed devices are not (yet) registered in the discovery service ([FreeAtHomeBindingDiscoveryService.java](https://github.com/ruebox/openhab2-addons/blob/feature_binding_freeathome/addons/binding/org.openhab.binding.freeathome/src/main/java/org/openhab/binding/freeathome/discovery/FreeAtHomeBindingDiscoveryService.java).

In order to add them, I would need 
* deviceId
* Device type: Raffstore, Switch, Temperature, Group (please refer to [FreeAtHomeBindingDiscoveryService.java](https://github.com/ruebox/openhab2-addons/blob/feature_binding_freeathome/addons/binding/org.openhab.binding.freeathome/src/main/java/org/openhab/binding/freeathome/discovery/FreeAtHomeBindingDiscoveryService.java))
* Each action of the device is linked to a channel and to a value (please refer to e.g. [FreeAtHomeRaffStoreHandler](https://github.com/ruebox/openhab2-addons/blob/feature_binding_freeathome/addons/binding/org.openhab.binding.freeathome/src/main/java/org/openhab/binding/freeathome/handler/FreeAtHomeRaffStoreHandler.java)

You could also use the debugging functionality of the binding:
* Dummy things: Also things are discovered. If a deviceId is not found, then a dummy item is generated: DiscoveryResult discoveryResult1 = DiscoveryResultBuilder                                   .create(uid1).withLabel("Dummy_" + device.DeviceDisplayName + "_" + device.DeviceTypeName + "_" + deviceTypeId + "_" + device.Serial)
* Debug log: An update.csv file is stored in the log directory that traces all events recognized by the binding. There you will find the output of the actions triggered of the devices itself or via web interface.

Would be great if you can provide the necessary info. I would introduce it with the next release.

**Comment by Kennys1710** [comment]
Hello, do you have an Update for wireless Aktoren like this:
Dummy_Sensor/Schalta. 4x Wohnen Terrasse links_Sensor/ Schaltaktor 2/2-fach_2039_ABB700CD4884
freeathome:dummy:ABB700CD4884:dummy_channel

or this
Dummy_Rolladen 1x Essen_Sensor/ Jalousieaktor 2/1-fach_2039_ABB700CCA41A
freeathome:dummy:ABB700CCA41A:dummy_channel

Thanks
Kenny

**Comment by ruebox** [comment]
@Kennys1710 
In general it would be feasible to add these two devices. However, I would need your support to test preliminary builds and to test it with your installation. Would this workout.

* Sensor/ Schaltaktor 2/2-fach_2039_ABB700CD4884: I would assume that this is a 2x switch with motion sensor. Is this correct?
* Sensor/ Jalousieaktor 2/1-fach_2039_ABB700CCA41A: Is this a motion sensor with 1x raffstore control (e.g. up / down , percentage)

Are you sure that the device Ids are identical. I assume that different devices have a different deviceId.

Thx. Best

**Comment by ruebox** [comment]
@mnielsm : Thanks for your feedback. However, I am surprised that only a few devices are discovered. 
According to the implementation the flowing situation look alike:
> * Thermostat (deviceID 1004)
Should be discovered (https://github.com/ruebox/openhab2-addons/blob/565d7d36477c9f8049d14036c25ab18c688a9e30/addons/binding/org.openhab.binding.freeathome/src/main/java/org/openhab/binding/freeathome/discovery/FreeAtHomeBindingDiscoveryService.java#L267)
> 
> * 2/2 sensor/rocker (deviceID 1010)
not supported, yet.
> * Motion sensor (deviceID 100A)
Can not be supported. motion sensors are not sending events via XMPP.
> * 1/1 sensor/dimmer (deviceID 1017)
not supported, yet
> * 1/1 sensor/rocker (deviceID 100C)
not supported, yet.
> * Wireless 1/1 sensor/dimmer (deviceID 2039)
not supported, yet.
> * 1 sensor (deviceID 1000)

Proposal: I could generate a new binary for testing purpose adding the listed deviceIds. If you verify the implementation, I could add these deviceIds to the offical release. What do you think?

**Comment by Kennys1710** [comment]
@ruebox 
Hello, thank you for your help.
Both actuators are not motion sensors. I'll post you the product page of Busch Jäger.
I have the following wireless actuators in use

https://www.busch-jaeger-katalog.de/index.php?bereich=1013622&programm=1065509

Sensor/Schaltaktor 2/2-fach, Wireless
Sensor/Jalousieaktor 2/1-fach, Wireless
Sensor/Jalousieaktor 1/1-fach, Wireless

If you need more information, please let me know

Many Thanks
Kenny

**Comment by KucingB** [comment]
@ruebox 

Hello, thanks for all the good work on the free@home binding.

I have installed your latest IoT_marketplace binding and its "online".

My setup:
F@H FW 2.2.5
OH2.2.4 in a Synology Docker container for 24/7.
OH2.2.4 on Windows 10 for test purposes.

When I trigger the discovery, I only finds 8 things:

Some of the 'supported' devices, are not found.
And in addition, there are a number of devices not yet supported.
Using the 'discover dummy' option, lists the "Dummy__" devices.

If I have to prioritize, I would like to get the Binäreingang 2-fach to work correctly first.

Example: Failed:
**Dummy_Binäreingang 2-fach, UP_B005** 
Dummy_Sensoreenheid 1-voudig_Sensoreinheit 1-fach_1000
Dummy_Sensoreenheid 2-voudig_Sensoreinheit 2-fach_1002
Dummy_Bewegingsschakelaar_Bewegungsmelder_1008
Dummy_Sensor/dimaktor 1/1-voudig_Sensor/Dimmaktor 1/1-fach_1017
Dummy_Sensor/dimaktor 2/1-voudig_Sensor/Dimmaktor 2/1-fach_1019
Dummy_Sensor/schakelaktor 1/1-voudig_Sensor/ Schaltaktor 1/1-fach_100C
Dummy_Sensor/schakelaktor 2/1-voudig_Sensor/ Schaltaktor 2/1-fach_100E
Dummy_Sensor/schakelaktor 2/2-voudig_Sensor/ Schaltaktor 2/2-fach_1010

Example: Found&recognized:
Hue color lamp 1_Hue Aktor_10C0 (== Dimmer)
Ruimtetemperatuurregelaar_Raumtemperaturregler_1004 (== Thermostat)
Verlichting_Neue Szene_4800 (== Scene)

If you can add these device ID's to the binary, I can help verifying.

Can you help?

Thanks.
